src/core/meshcop/tcat_agent.cpp

```diff
@@ -0,0 +1,524 @@
+            if (datasetInfo.IsNetworkNamePresent() && mCommissionerHasNetworkName &&
+                (datasetInfo.GetNetworkName() == mCommissionerNetworkName))
+            {
+                networkNamesMatch = true;
+            }
+
+            if (datasetInfo.IsExtendedPanIdPresent() && mCommissionerHasExtendedPanId &&
+                (datasetInfo.GetExtendedPanId() == mCommissionerExtendedPanId))
+            {
+                extendedPanIdsMatch = true;
+            }
+        }
+    }
+
+    if (!networkNamesMatch)
+    {
+        VerifyOrExit((aCommissionerCommandClassFlags & kNetworkNameFlag) == 0);
+    }
+    else if (aDeviceCommandClassFlags & kNetworkNameFlag)
+    {
+        additionalDeviceRequirementMet = true;
+    }
+
+    if (!extendedPanIdsMatch)
+    {
+        VerifyOrExit((aCommissionerCommandClassFlags & kExtendedPanIdFlag) == 0);
+    }
+    else if (aDeviceCommandClassFlags & kExtendedPanIdFlag)
+    {
+        additionalDeviceRequirementMet = true;
+    }
+
+#if (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
+    if (mCommissionerHasDomainName && (Get<MeshCoP::NetworkNameManager>().GetDomainName() == mCommissionerDomainName))
+    {
+        domainNamesMatch = true;
+    }
+#endif
+
+    if (!domainNamesMatch)
+    {
+        VerifyOrExit((aCommissionerCommandClassFlags & kThreadDomainFlag) == 0);
+    }
+    else if (aDeviceCommandClassFlags & kThreadDomainFlag)
+    {
+        additionalDeviceRequirementMet = true;
+    }
+
+    if (additionalDeviceRequirementMet)
+    {
+        authorized = true;
+    }
+
+exit:
+    return authorized;
+}
+
+bool TcatAgent::IsCommandClassAuthorized(CommandClass aCommandClass) const
+{
+    bool authorized = false;
+
+    switch (aCommandClass)
+    {
+    case kGeneral:
+        authorized = true;
+        break;
+
+    case kCommissioning:
+        authorized = IsCommandClassAuthorized(mCommissionerAuthorizationField.mCommissioningFlags,
+                                              mDeviceAuthorizationField.mCommissioningFlags, nullptr);
+        break;
+
+    case kExtraction:
+        authorized = IsCommandClassAuthorized(mCommissionerAuthorizationField.mExtractionFlags,
+                                              mDeviceAuthorizationField.mExtractionFlags, nullptr);
+        break;
+
+    case kTlvDecommissioning:
+        authorized = IsCommandClassAuthorized(mCommissionerAuthorizationField.mDecommissioningFlags,
+                                              mDeviceAuthorizationField.mDecommissioningFlags, nullptr);
+        break;
+
+    case kApplication:
+        authorized = IsCommandClassAuthorized(mCommissionerAuthorizationField.mApplicationFlags,
+                                              mDeviceAuthorizationField.mApplicationFlags, nullptr);
+        break;
+
+    case kInvalid:
+        authorized = false;
+        break;
+    }
+
+    return authorized;
+}
+
+TcatAgent::CommandClass TcatAgent::GetCommandClass(uint8_t aTlvType) const
```

Comment:
Ideally this should return an enum of type otTcatCommandClass .
The class is just the first 3 bits of `aTlvType`, for bitvalue <= 0b101. Anything higher is class 'invalid'.

The extra static definitions here aren't needed if we use otTcatCommandClass. See also column 1 in Table 16-3.

Reply:
resolved now